gunpowder-api/src/main/kotlin/io/github/gunpowder/api/util/Location.kt

```diff
@@ -27,27 +27,84 @@ package io.github.gunpowder.api.util
 import io.github.gunpowder.api.GunpowderMod
 import net.minecraft.entity.Entity
 import net.minecraft.server.world.ServerWorld
+import net.minecraft.util.BlockRotation
 import net.minecraft.util.Identifier
+import net.minecraft.util.math.Direction
 import net.minecraft.util.math.Vec2f
 import net.minecraft.util.math.Vec3d
 import net.minecraft.util.registry.Registry
 import net.minecraft.util.registry.RegistryKey
 import net.minecraft.world.World
 
 data class Location(
-        var position: Vec3d,
-        var rotation: Vec2f?,
-        var dimension: Identifier
+        val position: Vec3d,
+        val rotation: Vec2f?,
+        val dimension: Identifier
 ) {
     val world: ServerWorld?
         get() = GunpowderMod.instance.server.getWorld(RegistryKey.of(Registry.DIMENSION, dimension))
 
     companion object {
         @JvmStatic
-        fun of(entity: Entity) = Location(entity.pos, Vec2f(entity.yaw, entity.pitch), entity.world.registryKey.value)
+        val origin = Location(Vec3d.ZERO, Vec2f.ZERO, World.OVERWORLD.value)
```

Comment:
Not sure about this name, would like some more opinions on this. name should probably be capitalized.